src/components/Status/StatusForm.tsx

```diff
@@ -1,178 +1,178 @@
-import React, { useState } from 'react';
-import { StatusThresholds, RangeNames } from '../../services/firestore';
-import './StatusForm.css';
-
-interface StatusFormProps {
-  onSubmit: (name: string, thresholds: StatusThresholds) => Promise<void>;
-  onCancel: () => void;
-  loading?: boolean;
-}
-
-const StatusForm: React.FC<StatusFormProps> = ({ onSubmit, onCancel, loading }) => {
-  const [name, setName] = useState('');
-  const [normalThreshold, setNormalThreshold] = useState('');
-  const [elevatedThreshold, setElevatedThreshold] = useState('');
-  const [highThreshold, setHighThreshold] = useState('');
-  const [rangeNames, setRangeNames] = useState<RangeNames>({
-    normal: 'Normal',
-    elevated: 'Elevated',
-    high: 'High'
-  });
-  const [error, setError] = useState('');
-
-  const handleSubmit = async (e: React.FormEvent) => {
-    e.preventDefault();
-    setError('');
-
-    if (!name.trim()) {
-      setError('Please enter a status name');
-      return;
-    }
-
-    if (!rangeNames.normal.trim() || !rangeNames.elevated.trim() || !rangeNames.high.trim()) {
-      setError('Please enter names for all ranges');
-      return;
-    }
-
-    const normal = parseFloat(normalThreshold);
-    const elevated = parseFloat(elevatedThreshold);
-    const high = parseFloat(highThreshold);
-
-    if (isNaN(normal) || isNaN(elevated) || isNaN(high)) {
-      setError('Please enter valid numbers for all thresholds');
-      return;
-    }
-
-    if (normal >= elevated || elevated >= high) {
-      setError('Thresholds must be in ascending order: First < Second < Third');
-      return;
-    }
-
```

Comment:
Using a form library here might interesting, since you got some complex validations.